microsetta_interface/implementation.py

```diff
@@ -1410,10 +1411,22 @@ def get_fill_source_survey(*,
                     ctr += 1
                     survey_question_count += 1
                     field['label'] = str(ctr) + ". " + field['label']
-                field['label'] = '<span class="survey-skip small-text" '\
-                                 + 'onClick="skipQuestion(this)">'\
-                                 + gettext('SKIP')\
-                                 + '</span>' + field['label']
+
+                if need_reconsent:
+                    # If the user has not agreed to the current consent, we
+                    # disable all of the fields.
+                    field['disabled'] = True
```

Comment:
Out of paranoia, and the fun you can have in a javascript console, do we assert consent prior to storing in the backend? I think yes but just checking...

Reply:
Yes - https://github.com/biocore/microsetta-interface/blob/csymons_production_feedback/microsetta_interface/implementation.py#L1455